### ISSUE TITLE
BUG: Compass EVM failed to deploy on all chains except Base after Vote 29

### ISSUE THREAD

**Issue by taariq** [VolumeFi/paloma#778]
# What is happening?
<details>
  <summary>Section description</summary>
<i>Provide as much context as you can. Give as much context as you can to make it easier for the developers to figure what is happening.</i>
</details>

Vote on Mainnet to Deploy Compass passes: https://paloma.explorers.guru/proposal/29

|Compass Deploy Status | Chain Name | Compass Contract
|------------- | ------------- | ------------- |
| ✅ | Base | https://basescan.org/address/0xcBbADBD4aAdA4Db509F3ba80E69E37bFB793e333|
| ❌ | BNB |
| ❌ | ETH |
| ❌ | Matic |
| ❌ | OP | 

**from @verabehr** 
1. Confirmed that we don't have any errors because our pigeon wasn't selected to deploy any of these messages --> if we do this again, we should set up a script to watch the queue before the vote passes
2. We don't have any new attempts yet, but I think we will have some, because it tries over and over. Last time it took a whole night (many hours)
3. It's definitely not in the queue right now, attempting to deploy the contract is part of some paloma loop, so what should happen is that the contract deployment will be put back into the queue where the message sending is retried and then if it fails the message is removed from teh queue, and then it will eventually put back into the queue again.
4. **Speculation:** I'm not 100% what the team consensus was last time, somehow the notion doesn't have the conclusion, but I think this is because a jailed pigeon picks up the message or a pigeon that is somehow not able to deliver the deployment e.g. due to lack of funds

# Paloma and pigeon versions and logs
<details>
  <summary>Section description</summary>
<i>Write down paloma version.</i>
<i>Write down pigeon version.</i>
<i>Copy and paste pigeon config file as well as relevant ENV variables.</i>
</details>

Paloma 1.8.0 and Pigeon 1.8.1

**2 hours of logs in the root directory of vadmin user**
`palomad_9-14.txt` and `pigeond_9-14.txt`

**script and files to observe queues in root directory of root user**
`message_check.sh`  `message_queue_bnb.json`  `message_queue_eth.json`  `message_queue.json`  `message_queue_matic.json`  `message_queue_op.json`

# How to reproduce?
<details>
  <summary>Section description</summary>
  <i>Please write detailed steps of what you were doing for this bug to appear.</i>
</details>

Deploy Compass-EVM to multiple chains ALREADY deployed with Compass-EVM

# What is the expected behaviour?
<details>
  <summary>Section description</summary>
<i>If you know, please write down what is the expected behaviour. If you don't know, that's ok. We can have a discussion in comments.</i>
</details>

Compass-EVM will redeploy to all chains after

**Comment by byte-bandit** [comment]
From a first impression, this is the same that happened last week. Most likely the deployments failed due to Pigeons missing funds or another issue.

I have refreshed the deployment for the missing chains, and each update went through on the first attempt. These are the latest compass addresses:

|Compass Deploy Status | Chain Name | Compass Contract
|------------- | ------------- | ------------- |
| ✅ | Base | https://basescan.org/address/0xcBbADBD4aAdA4Db509F3ba80E69E37bFB793e333|
| ❌ | BNB | https://bscscan.com/address/0x79e9f573D908E012aC6E1F7978cb09e0065E6355 |
| ❌ | ETH | https://etherscan.io/address/0xa00CF3504FfaCff301E4973e21B44C052d087157 |
| ❌ | Matic | https://polygonscan.com/address/0x92c5D97a2069b9cbC117d5Df9eA119fc635a9319 |
| ❌ | OP | https://optimistic.etherscan.io/address/0xe3e2E3ebd228485FCbC812D1DBA0B2F90233E56c |

I will take a look at the logs and follow up on what happened.

**Comment by byte-bandit** [comment]
Taking a look at `eth-main` as an example here:

```
Sep 14 19:28:25 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","queue-type-name":"evm/eth-main/evm-turnstone-message","message-id":98798,"time":"2023-09-14T19:28:25Z","message":"put message into consensus queue"}
Sep 14 19:28:25 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/evm","chain-reference-id":"eth-main","smart-contract-id":4,"time":"2023-09-14T19:28:25Z","message":"added a new smart contract deployment to queue"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x3A188DE3d16ddee783990dDEd1B629E95C8242d9","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x377D23948D41579f2c3cA40308e3bdd53f6dA7B2","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x43E218f96A567DC26C6e65fCabF1fDE26Af69444","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x4a89f96fdff3C161937CFBC3e22d2E325612aaEc","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x732fBb6018F2cB5b844055C9EB567447a3132833","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x19f5911E4cA69E30449aD6BB71De341F01F118BB","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x0DAaBB4FF60423Eb1F14cC6731394e098ad51bcb","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x1Dd71EE9e7b1CfF10CCb7E12520999c2d4fB45e5","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x3aB643728bD503AE41561B4Bfe1D309A17255b8D","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0xC47b85c7577260d995BE7aB65728AE33B4CD26cD","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0xeE83770168D4EE756e74562C17BDED88091519D0","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x7C3dE83D90E8d2001a7e3F38E6a7058334cB291E","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x1F6EF2784cbB2A5011d8CF12356fb5ee3eAeF372","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x092604a080524Eff1C875fF13A1b88cBDD41df37","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x63f55bc560E981d53E1f5bb3643e3a96D26fc635","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:30 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0xDEea5b069208e0eE37b630A3e7672FC50e8fE24a","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:30Z","message":"added message signature"}
Sep 14 19:28:32 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0xDEea5b069208e0eE37b630A3e7672FC50e8fE24a","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:32Z","message":"added message signature"}
Sep 14 19:28:32 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x443266026738061972012e62D5Ecd9D98da8B6F4","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:32Z","message":"added message signature"}
Sep 14 19:28:32 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x1ad90dB98da083E117D5F62a1673fC0f3A5930ca","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:32Z","message":"added message signature"}
Sep 14 19:28:32 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x5A7C9FC846b18944FD2B8F8aE7255dB5af7b9c04","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:32Z","message":"added message signature"}
Sep 14 19:28:32 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0xEe21301aF1d9562B5cBEdf520077Ea0a9bC9d535","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:32Z","message":"added message signature"}
Sep 14 19:28:32 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0xAAc74d38c82c367b3dA7482E3Aecf6DD7A512eF2","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:32Z","message":"added message signature"}
Sep 14 19:28:33 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x443520951d766456F2AC5578f3E7e4B56a7BcF63","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:33Z","message":"added message signature"}
Sep 14 19:28:33 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0xADe5F5eFeAe72102d998F6C496ea59731a70eEb0","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:33Z","message":"added message signature"}
Sep 14 19:28:33 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x279fC7c0ebEf3328d3899FE73464347A764C5f2C","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:33Z","message":"added message signature"}
Sep 14 19:28:35 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x6dc59EE4bdFa2C791229004f29b08F783491a934","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:35Z","message":"added message signature"}
Sep 14 19:28:35 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0xeB784b37365C302C97C9Ec8cB0933cE344e6dE42","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:35Z","message":"added message signature"}
Sep 14 19:28:39 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:39Z","message":"added message public access data"}
Sep 14 19:28:41 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0x179206Cd9080A8ee77C2256A27050112da13b2f6","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:41Z","message":"added message signature"}
Sep 14 19:28:50 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","message-id":98798,"queue-type-name":"evm/eth-main/evm-turnstone-message","signed-by-address":"0xBBEFE691d2C3bbB835CE2958b453ccd05BdB27A5","chain-type":"evm","chain-reference-id":"eth-main","time":"2023-09-14T19:28:50Z","message":"added message signature"}
Sep 14 19:31:39 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","queue-type-name":"evm/eth-main/validators-balances","message-id":98803,"time":"2023-09-14T19:31:39Z","message":"put message into consensus queue"}
Sep 14 19:39:22 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","queue-type-name":"evm/eth-main/validators-balances","message-id":98808,"time":"2023-09-14T19:39:22Z","message":"put message into consensus queue"}
Sep 14 19:47:05 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","queue-type-name":"evm/eth-main/validators-balances","message-id":98813,"time":"2023-09-14T19:47:05Z","message":"put message into consensus queue"}
Sep 14 19:54:50 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","queue-type-name":"evm/eth-main/validators-balances","message-id":98818,"time":"2023-09-14T19:54:50Z","message":"put message into consensus queue"}
Sep 14 20:02:36 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","queue-type-name":"evm/eth-main/validators-balances","message-id":98823,"time":"2023-09-14T20:02:36Z","message":"put message into consensus queue"}
Sep 14 20:10:21 mainnet-validator palomad[237038]: {"level":"info","module":"server","module":"x/palomaconsensus","queue-type-name":"evm/eth-main/validators-balances","message-id":98828,"time":"2023-09-14T20:10:21Z","message":"put message into consensus queue"}
```

You can see the new compass EVM contract being added to the message queue, followed by all Pigeons picking it up for signing. 

Interestingly, towards the bottom, you can see the log message `added message public access data`, which indicates that the process finished successfully. There's also some additional logging below showing that more messages where added to the queue, but those are just balance updates.

This is the same pattern for every other chain, **including** `base`, which actually finished successfully. I have not found any other suspicious lines in the log, so from what we can take away from this, everything LOOKS like it went fine. It didn't Not good.

## TL;DR

There's nothing in the logs indicating anything went wrong at all. We cannot make any further deductions from this without access to what actually happened to the message. Since messages are deleted from Paloma once they've been handled, there's no way of knowing.

## Action items

I propose we also add `public access data` and `error data` to the log. That way, this information will no longer be lost to us after the message disappears and we'll be a bit smarter again. Honestly, I have no explanation why this isn't log in the first place. The only reason I can think of are security concerns, but then, all this information is accessible to all nodes already anyway while the message is still on the queue, so really I don't see a reason not to.

@taariq @verabehr Your thoughts on this?

**Comment by taariq** [comment]
Let's log it @byte-bandit. Agreed.

**Comment by taariq** [comment]
Fixed with https://github.com/palomachain/paloma/pull/980.
